serf/internal_query.go

```diff
@@ -152,7 +159,8 @@ func (s *serfQueries) handleConflict(q *Query) {
 
 func (s *serfQueries) keyListResponseWithCorrectSize(q *Query, resp *nodeKeyResponse) ([]byte, messageQueryResponse, error) {
 	warned := false
-	for i := len(resp.Keys); i >= 0; i-- {
+	maxListKeys := q.serf.config.QueryResponseSizeLimit / maxListKeyFactor
```

Comment:
this is a crazy edge case but if `QueryResponseSizeLimit` was set to something less that `maxListKeyFactor` then `maxListKeys` would end up being `0` and the loop below would never run and we'd return...

Oh OK it returns an error at the bottom so it's fine I thought it was going to be an empty message. Fine!